### PR TITLE
Allow st2web to serve HTTPS

### PR DESCRIPTION
**Pull request by cognifloyd**
The Docker image allows serving HTTPS when environment var `ST2WEB_HTTPS=1`. Setting `ST2WEB_HTTPS: "1"` in `st2web.env`, however, breaks the `st2web` k8s deployment & service because the `livenessProbe` and the service are still trying to access HTTP on port 80.

To make `ST2WEB_HTTPS=1` work correctly, 2 things are required:
1. change the protocol and port if `ST2WEB_HTTPS=1`.
2. add `extra_volumes` (like several other deployments) so users can mount ssl certs that they provide. Managing any kubernetes secrets for those certs is out-of-scope for this chart, similar to anything else provided via `extra_volumes`.

This PR makes those changes, but retains the current HTTP-by-default behavior.

Closes: #236

### CODE REVIEW

**Review by ericreeves — APPROVED** [review]
Elegantly introduced feature, and backward-compatible.  This is a solid feature for those running in stacks who can't just shovel the work off to a cloud load balancer.

File: values.yaml [hunk 1]
```diff
@@ -295,8 +295,15 @@ st2web:
   affinity: {}
   env: {}
   # HTTP_PROXY: http://proxy:1234
+  ## Advanced: To configure st2web to serve HTTPS (port 443) instead of HTTP (port 80)
+  ## provide ssl certs via extra_volumes, and add the ST2WEB_HTTPS env var here:
+  # ST2WEB_HTTPS: 1
```
Comment (armab) [inline values.yaml:300]: It makes sense to quickly mention in the comments about the alternatives like LoadBalancer or Ingress Controller, which would be a more K8s-native approach to this problem.

**Review by armab — APPROVED** [review]
I thought about discussing more the approach in this PR as we've removed it before in favor of LoadBalancer or Ingress Controller which is more K8s native and flexible.

Because we have this functionality still present in the Docker images and enabling it really easy, - let's do it! 

Thanks for the PR :+1:

File: values.yaml (hunk 1 above)
Comment (cognifloyd) [reply values.yaml]: done

**Pull request merged**